src/components/Button/Button.tsx

```diff
@@ -17,22 +18,26 @@ export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
       fixedIcon: _fixedIcon,
       suffixIcon: _suffixIcon,
       type = 'button',
+      disabled = false,
+      loading = false,
       ...props
     },
     ref,
   ) => {
-    const icon = _icon === 'default' ? <VariantIcon variant={variant} /> : _icon;
+    const icon = loading ? <CircularProgress /> : _icon === 'default' ? <VariantIcon variant={variant} /> : _icon;
     const fixedIcon = _fixedIcon === 'default' ? <VariantIcon variant={variant} /> : _fixedIcon;
     const suffixIcon = _suffixIcon === 'default' ? <VariantIcon variant={variant} /> : _suffixIcon;
     const cls = clsx({
       [styles.button]: true,
       [styles[variant]]: true,
       [styles[size]]: true,
       [styles.block]: block,
-      [styles.disabled]: !!props.disabled,
+      [styles.disabled]: disabled,
+      [styles.loading]: loading,
     });
+
     return (
-      <button type={type} className={cls} ref={ref} {...props}>
+      <button type={type} className={cls} ref={ref} disabled={disabled || loading} {...props}>
```

Comment:
I'll have to test it, but if it is disabled, the screen reader will not be able to access the element (the button suddenly disappears from the screen?). 
It may not be possible to tell that it is on loading.

An idea instead is aria-disable="true" and stop the execution of onClick. What do you think?

https://azukiazusa.dev/blog/use-aria-disabled-to-give-focus-to-disabled-button/

Reply:
It may be possible to respond with another PR. I cannot determine the right solution.

Reply:
This article would be good reference. https://www.bekk.christmas/post/2023/24/accessible-loading-button

Reply:
But it depends on how we (as Ubie inc) think the button should be used and ideal expressions.

Reply:
@kodai3 

Thank you for sharing the article!

> t is only a short time and the next page, popup or something else should happen.

Depending on various cases, it can be assumed that not short

- Poor communication environment and prolonged loading
- The basic speed of the line used by the user is slow
- Loading continues to dance without errors being handled correctly (in this case, we are hopeful that the user can guess the situation if the loading trigger remains).
- Loading buttons are used for long processes (incorrect usage not assumed).

There also seems to be a problem that being disabled causes the focus to jump to unexpected places.

> If the button is the current item in focus when you add the disabled attribute (for example, if someone just pressed it to submit the form), the element actually loses focus, which shifts to the document element.

https://gomakethings.com/dont-disable-buttons/

React Spectrum, which focuses on accessibility, uses aria-disabled.
https://react-spectrum.adobe.com/react-spectrum/Button.html#pending

This is going to be a long discussion, so I'll leave the decision to @takanorip ! 😇

Reply:
We should use `aria-disabled`.
Since the loading time can vary depending on the user's environment, I believe it is better to lean towards the safer side in implementation. 
Accessibility is an important element in Ubie UI, so we should aim for a more accessible implementation.